### PR TITLE
Add support to cache on specific json keys

### PR DESCRIPTION
**Pull request by samchungy**
Addressing: https://github.com/seek-oss/skuba/pull/1512

Tested on one of my repos:

Cache hit in pipeline while changing package.json
https://github.com/SEEK-Jobs/hirer-budgets/compare/eb9853f8c9cc4aa4939faaf9ca764148dc351aa8...b0d8c4db2e9a6e501a51a52356c41987b1de8b53

### CODE REVIEW

**Review by AaronMoat — APPROVED** [review]
I can't spot anything wrong, but 
image

File: README.md [hunk 1]
```diff
@@ -76,6 +76,23 @@ steps:
...
+steps:
+  - command: npm test
+    plugins:
+      - seek-oss/docker-ecr-cache#v2.2.0:
+          cache-on:
+            - package.json#.dependencies
+            - package.json#.devDependencies
+            - package.json#.pnpm.overrides
+            - yarn.lock
+      - docker#v3.12.0:
```
Comment (72636c) [inline README.md:95]: Might be time to renovate these references 😅

File: README.md [hunk 2]
```diff
@@ -76,6 +76,23 @@ steps:
             - /workdir/node_modules
 ```
 
+It also supports caching on specific JSON keys which can be specified following a `#` character using [jq syntax](https://jqlang.github.io/jq/manual/#object-identifier-index). This requires [jq](https://jqlang.github.io/jq/) to be installed on the build agent. You cannot use this in conjunction with Bash globbing.
```
Comment (72636c) [inline README.md]: More for our internal understanding, is it worth clarifying

- The implementation matches on the first `.json#` substring, which would be silly to have elsewhere in your file path but is nonetheless possible on Unix systems
- A given entry cannot mix globbing and jq paths, but you can do `['**/LICENSE.md', 'package.json#.dependencies']`

**Review by 72636c — APPROVED** [review]
👏

File: README.md [hunk 3]
```diff
@@ -76,6 +76,23 @@ steps:
...
+```yaml
+steps:
+  - command: npm test
+    plugins:
+      - seek-oss/docker-ecr-cache#v2.2.0:
+          cache-on:
+            - package.json#.dependencies
+            - package.json#.devDependencies
+            - package.json#.pnpm.overrides
+            - yarn.lock
```
Comment (72636c) [inline README.md]: Would it be useful to list our recommended set of cache keys for a typical pnpm project here?

```suggestion
  - command: pnpm install --offline && pnpm test
    plugins:
      - seek-oss/docker-ecr-cache#v2.2.0:
          cache-on:
            - .npmrc
            - package.json#.dependencies
            - package.json#.devDependencies
            - package.json#.packageManager
            - package.json#.pnpm.overrides
            - pnpm-lock.yaml
```

File: README.md (hunk 1 above)
Comment (samchungy) [reply README.md:95]: I'll tweak them in a separate PR, too many changes otherwise

**Pull request merged**